pkg/compose/provider.go

```diff
@@ -59,8 +62,43 @@ func NewLocalBlobProvider(fileProvider content.Store) BlobProvider {
 	}
 }
 
+type tokenProxyTripper struct {
+	base http.RoundTripper
+	host *url.URL
+}
+
+func (t tokenProxyTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	reqBodyClosed := false
+	if req.Body != nil {
+		defer func() {
+			if !reqBodyClosed {
+				req.Body.Close()
+			}
+		}()
+	}
+
+	req2 := req.Clone(req.Context())
+	req2.URL = t.host.JoinPath(req2.URL.Path)
+	req2.Host = t.host.Host
+
+	// req.Body is assumed to be closed by the base RoundTripper.
+	reqBodyClosed = true
+	return t.base.RoundTrip(req2)
+}
+
 func NewRemoteBlobProviderFromConfig(config *Config) BlobProvider {
 	client := NewHttpClient(config.ConnectTimeout, config.ReadTimeout)
+	if config.ProxyURL != nil {
+		tpt := tokenProxyTripper{
+			base: client.Transport,
+			host: config.ProxyURL,
+		}
+
+		if config.ProxyCerts != nil {
```

Comment:
Are `config.ProxyCerts` optional? If not, then we probably should check if it is set during the config parsing&setting phase.

Reply:
See my other comment - we could make them optional. It depends what we really want.